### PR TITLE
AEA-623 Expose crypto objects to the Connection class.

### LINKED ISSUE #1191: Wallet refactoring (split)
**Is your feature request related to a problem? Please describe.**
- Currently, the wallet considers all keys to be used by the agent
- Some are used by connection; these should not be the same used for econ/finanical exchange

**Describe the solution you'd like**
- Allow for the wallet to be appropriately configured to have key-pairs for dm and key-pairs for connections.
- Extend loading mechanisms of connections accordingly

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR exposes crypto objects to the `Connection` class.

## Fixes

Fix #1191 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -823,42 +822,6 @@ def _check_consistent(self, ledger_apis: LedgerApis) -> None:
             ledger_apis.default_ledger_id == self._default_ledger
         ), "Default ledger id of LedgerApis does not match provided default ledger."
 
-    # TODO: remove and replace with a clean approach (~noise based crypto module or similar)
-    def _update_identity(
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:827]: @marcofavorito please delete the p2p_noise package entirely from the packages dir

Comment (DavidMinarsch) [reply aea/aea_builder.py:827]: Or we can leave it if you feel like leaving it in.

Comment (DavidMinarsch) [reply aea/aea_builder.py:827]: But we don't support it so it is probably best deleting it

Comment (marcofavorito) [reply aea/aea_builder.py:827]: ok

File: aea/aea_builder.py [hunk 2]
```diff
@@ -771,11 +770,9 @@ def build(
         ledger_apis = self._load_ledger_apis(ledger_apis)
         self._load_and_add_components(ComponentType.PROTOCOL, resources)
         self._load_and_add_components(ComponentType.CONTRACT, resources)
-        connections = self._load_connections(identity.address, connection_ids)
-        identity = self._update_identity(identity, wallet, connections)
         aea = AEA(
             identity,
-            connections,
+            [],
```
Comment (marcofavorito) [inline aea/aea_builder.py:775]: to fix?

File: aea/aea_builder.py [hunk 3]
```diff
@@ -1240,10 +1209,23 @@ def _load_connection(
...
 
+    def _load_and_add_connections(
+        self,
+        aea: AEA,
+        wallet: Wallet,
+        connection_ids: Optional[Collection[PublicId]] = None,
+    ):
+        connections = self._load_connections(aea.identity, wallet, connection_ids)
+        for c in connections:
+            aea.multiplexer.add_connection(c, c.public_id == self._default_connection)
```
Comment (marcofavorito) [inline aea/aea_builder.py:1227]: @DavidMinarsch it is ok for now, but the connection could be:

- added to `resources` at build time
- retrieved from resources and added to the multiplexer on start

**Pull request merged**